src/main/java/net/glowstone/dispenser/ArmorDispenseBehavior.java

```diff
@@ -0,0 +1,50 @@
+package net.glowstone.dispenser;
+
+import java.util.ArrayList;
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.blocktype.BlockDispenser;
+import org.bukkit.block.BlockFace;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.Location;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Entity;
+import java.util.List;
+import org.bukkit.entity.Player;
+import net.glowstone.inventory.GlowInventory;
+
+public class ArmorDispenseBehavior extends DefaultDispenseBehavior {
+    DefaultDispenseBehavior defaultBehavior = new DefaultDispenseBehavior();
+    
+    @Override
+    protected ItemStack dispenseStack(GlowBlock block, ItemStack stack) {
+        BlockFace facing = BlockDispenser.getFacing(block);
+        GlowBlock target = block.getRelative(facing);
```

Comment:
It is a bit expensive to get a block, so try cloning `block`'s location and then adding facing.getModX,Y and Z to that location to get `targetLocation`.